vmdk_plugin/utils/refcount/refcnt.go

```diff
@@ -114,8 +124,15 @@ type refCount struct {
 
 // RefCountsMap struct
 type RefCountsMap struct {
+	// does this need to be atomic?
+	// this is just one time state transition
+	refcntSuccess bool
+
 	refMap map[string]*refCount // Map of refCounts
 	mtx    *sync.RWMutex        // Synchronizes RefCountsMap ops
+
+	auxRefMap  map[string]*refCount // do refcount
+	curr_state *refcount_state
```

Comment:
Rather than nesting, I would just get rid of refcount_state and add isDirty and dirtyMtx to RefCountsMap structure